liota/dccs/iotcv2.py

```diff
@@ -0,0 +1,346 @@
+        # block until there is an item available or timeout
+        try:
+            on_response(set_prop_resp_q.get(True, timeout), set_prop_resp_q)
+        except:
+            raise Exception("Exception while setting properties")
+
+        if not self.set_prop:
+            raise SetPropertiesFailure()
+        log.info("Set Properties for {0} succeeded".format(entity.name))
+
+    def unregister(self, entity_obj):
+        pass
+
+    def _registration(self, operation_id, guid, mo_name, mo_type):
+        log.info("operation_id:{0} guid:{1}".format(operation_id, guid))
+        return {
+            "type": "registerMO_req",
+            "guid" : guid,
+            "name" : mo_name,
+            "MOType" : mo_type,
+            "operation_id" : operation_id
+        }
+
+    def _properties(self, operation_id, mo_type, guid, mo_name, properties):
+        msg = {
+            "type": "setProperties_req",
+            "guid" : guid,
+            "name" : mo_name,
+            "MOType" : mo_type,
+            "operation_id" : operation_id,
+            "properties": []
+        }
+        for key, value in properties.items():
+            msg["properties"].append({"key": key, "value": value})
+        return msg
+
+    def _relationship(self, operation_id, parent_res_uuid, child_res_uuid):
+        return {
+            "type": "createRelationships_req",
+            "operation_id" : operation_id,
+            "parent": parent_res_uuid,
+            "child": child_res_uuid
+        }
+
+    def next_id(self):
+        self.counter = (self.counter + 1) & 0xffffff
+        # Enforce even IDs
+        return int(self.counter * 2)
+
+    def _dispatch_recvd_msg(self):
+        log.info("Dispatching received messages from IoTCV2")
+
+        while True:
+            try:
+                # block until there is an item available
+                msg = self.recv_msg_queue.get(True)
+                log.debug("Received msg: {0}".format(msg))
+                json_msg = json.loads(msg)
+                log.debug("Processing msg: type:{0} operation_id:{1}".format(json_msg["type"], json_msg["operation_id"]))
+                # search matched request in request dictionary
+                # assume operation_id will be unique in one process
+                log.debug("self.req_dict {0} keys:{1}".format(self.req_dict, self.req_dict.keys()))
+                if (json_msg["operation_id"] in self.req_dict.keys()):
+                    req = self.req_dict[json_msg["operation_id"]]
+                    log.debug("self.req_dict {0} req:{1}".format(self.req_dict, req))
+                    # get/delete request from dictionary
+                    if (req is not None):
+                        log.debug("json_msg['guid'] {0} req.guid:{1}".format(json_msg["guid"], req.guid))
+                        # double confirm guid
+                        if (json_msg["guid"] != req.guid):
```

Comment:
why are we confirming guid here, and then again inside on_response?

Reply:
as comment said, double confirm to guarantee between there is not disruption or destroy.